File: paciente.py
import csv
import pandas as pd


class Paciente:
    def __init__(self, Id):
        self.ID = Id
        self.ruta = "Archivos/pacientes.csv"
        df = pd.read_csv(self.ruta)
        # Buscar la fila donde el ID coincide
        df['ID'] = df['ID'].astype(str)
        self.ID = str(self.ID)

        
        paciente_data = df[df['ID'] == self.ID]
        if not paciente_data.empty:
            # Asignar valores a los atributos de la instancia
            self.nombre = paciente_data.iloc[0]['Nombre']
            self.edad = paciente_data.iloc[0]['Edad']
            self.diagnostico = paciente_data.iloc[0]['Diagnóstico']
            self.especialidad = paciente_data.iloc[0]['EspecialidadAsignada']
            self.cama_asignada = paciente_data.iloc[0]['CamaAsignada']
            self.fecha_ingreso = paciente_data.iloc[0]['FechaIngreso']
            self.historial_medico = paciente_data.iloc[0]['HistorialMedico']

    def consultarPaciente(self):
        """Mostrar la información del paciente de la instancia actual."""
        if self.nombre:
            print(f"Información del Paciente:")
            print(f"ID: {self.ID}")
            print(f"Nombre: {self.nombre}")
            print(f"Edad: {self.edad}")
            print(f"Diagnóstico: {self.diagnostico}")
            print(f"Especialidad Asignada: {self.especialidad}")
            print(f"Cama Asignada: {self.cama_asignada}")
            print(f"Fecha de Ingreso: {self.fecha_ingreso}")
            print(f"Historial Médico: {self.historial_medico}")

    def consultar_historial(self):
        print(self.historial_medico)

    # Método para cambiar la especialidad
    def cambiar_especialidad(self, especialidad_nueva):
        self.especialidad = especialidad_nueva
                
        # Leer el archivo CSV
        df = pd.read_csv(self.ruta)

        # Asegurarse de que ID y la columna 'ID' tienen el mismo tipo
        df['ID'] = df['ID'].astype(str)
        self.ID = str(self.ID)

        # Verificar si el ID existe en el archivo CSV
        if self.ID in df['ID'].values:
            # Localizar la fila del paciente con el ID y cambiar la especialidad
            df.loc[df['ID'] == self.ID, 'EspecialidadAsignada'] = especialidad_nueva
            
            # Guardar el DataFrame actualizado en el archivo CSV
            df.to_csv(self.ruta, index=False)
            print("La especialidad se cambió con éxito.")
        else:
            print(f"No se encontró el ID {self.ID} en el archivo CSV.")

File: Csharp/HealthTech/Paciente.cs

namespace HealtTech
{
    using System;
    using System.Collections.Generic;
    using System.IO;
    using System.Linq;
    using CsvHelper;
    using CsvHelper.Configuration;
    using System.Globalization;

    public class Paciente
    {
        public string ID { get; set; }
        public string Nombre { get; set; }
        public string Edad { get; set; }
        public string Diagnostico { get; set; }
        public string EspecialidadAsignada { get; set; }
        public string CamaAsignada { get; set; }
        public string FechaIngreso { get; set; }
        public string HistorialMedico { get; set; }
        private string ruta;

        public Paciente(string id)
        {
            ID = id;
            ruta = @"C:\Users\Luis Angel Seoanes O\OneDrive\Escritorio\HealtTech\Csharp\Archivos\pacientes.csv";

            // Leer el archivo CSV con CsvHelper
            using (var reader = new StreamReader(ruta))
            using (var csv = new CsvReader(reader, new CsvConfiguration(CultureInfo.InvariantCulture)))
            {
                var pacientes = csv.GetRecords<PacienteData>().ToList();

                // Buscar el paciente por ID
                var paciente = pacientes.FirstOrDefault(p => p.ID == ID);

                if (paciente != null)
                {
                    // Asignar los valores a los atributos de la instancia
                    Nombre = paciente.Nombre;
                    Edad = paciente.Edad;
                    Diagnostico = paciente.Diagnostico;
                    EspecialidadAsignada = paciente.EspecialidadAsignada;
                    CamaAsignada = paciente.CamaAsignada;
                    FechaIngreso = paciente.FechaIngreso;
                    HistorialMedico = paciente.HistorialMedico;
                }
            }
        }

        public void ConsultarPaciente()
        {
            // Mostrar la información del paciente
            if (!string.IsNullOrEmpty(Nombre))
            {
                Console.WriteLine("Información del Paciente:");
                Console.WriteLine($"ID: {ID}");
                Console.WriteLine($"Nombre: {Nombre}");
                Console.WriteLine($"Edad: {Edad}");
                Console.WriteLine($"Diagnóstico: {Diagnostico}");
                Console.WriteLine($"Especialidad Asignada: {EspecialidadAsignada}");
                Console.WriteLine($"Cama Asignada: {CamaAsignada}");
                Console.WriteLine($"Fecha de Ingreso: {FechaIngreso}");
                Console.WriteLine($"Historial Médico: {HistorialMedico}");
            }
        }

        public void ConsultarHistorial()
        {
            Console.WriteLine(HistorialMedico);
        }

        public void CambiarEspecialidad(string especialidadNueva)
        {
            // Cambiar la especialidad del paciente
            EspecialidadAsignada = especialidadNueva;

            // Leer el archivo CSV
            var pacientes = new List<PacienteData>();
            using (var reader = new StreamReader(ruta))
            using (var csv = new CsvReader(reader, new CsvConfiguration(CultureInfo.InvariantCulture)))
            {
                pacientes = csv.GetRecords<PacienteData>().ToList();
            }

            // Verificar si el ID existe en el archivo CSV
            var paciente = pacientes.FirstOrDefault(p => p.ID == ID);
            if (paciente != null)
            {
                // Cambiar la especialidad
                paciente.EspecialidadAsignada = especialidadNueva;

                // Guardar el DataFrame actualizado en el archivo CSV
                using (var writer = new StreamWriter(ruta))
                using (var csv = new CsvWriter(writer, new CsvConfiguration(CultureInfo.InvariantCulture)))
                {
                    csv.WriteRecords(pacientes);
                }

                Console.WriteLine("La especialidad se cambió con éxito.");
            }
            else
            {
                Console.WriteLine($"No se encontró el ID {ID} en el archivo CSV.");
            }
        }
    }

    // Clase auxiliar para representar los datos del archivo CSV
    public class PacienteData
    {
        public string ID { get; set; }
        public string Nombre { get; set; }
        public string Edad { get; set; }
        public string Diagnostico { get; set; }
        public string EspecialidadAsignada { get; set; }
        public string CamaAsignada { get; set; }
        public string FechaIngreso { get; set; }
        public string HistorialMedico { get; set; }
    }

}
